packages/js/src/settings/helpers/submit.js

```diff
@@ -3,6 +3,45 @@ import { __ } from "@wordpress/i18n";
 import { forEach, get, includes, isArray, isObject, omit } from "lodash";
 import { STORE_NAME } from "../constants";
 
+/**
+ * @param {string} responseText The responseText.
+ * @returns {void}
+ */
+const handleLlmstxtFailure = ( responseText ) => {
+	const { setGenerationFailure } = dispatch( STORE_NAME );
+
+	if ( includes( responseText, "{{ yoast-llms-txt-generation-failure: " ) ) {
+		// We do a nested if here for performance reasons, so that we add a single `includes` operation and not more, in most cases.
+		if ( includes( responseText, "{{ yoast-llms-txt-generation-failure: filesystem_permissions }}" ) ) {
+			setGenerationFailure( {
+				generationFailure: true,
+				generationFailureReason: "filesystem_permissions",
```

Comment:
It might be good to put the reason strings constants, isolate them in a dedicated file and use the constants instead.